### ISSUE TITLE
The TS compiler failed to modify the JSDoc node

### ISSUE THREAD

**Issue by Binote** [microsoft/TypeScript#47097]
# Bug Report

No errors reported

### 💻 Code

```ts
import * as ts from 'typescript'
const anExampleVariable = "Hello World"
console.log(anExampleVariable)

// To learn more about the language, click above in "Examples" or "What's New".
// Otherwise, get started by removing these comments and the world is your playground.


function extract() {

    const printer = ts.createPrinter();
    const sourceFile = ts.createSourceFile('test.ts', `
/**
 * 数字相加
 * @param  a 数字aa
 * @param  b 数字b
 * @returns  a+b的结果
 */
function add2(a:number,b:number):number {
  return a+b
}
    `, ts.ScriptTarget.Latest, false, ts.ScriptKind.TS)
    travers(sourceFile,
      /**
       *
       * @param {ts.FunctionDeclaration} node
       */
      (node)=>{
      switch (node.kind){
        case ts.SyntaxKind.FunctionDeclaration:
            const jsDocs = node.jsDoc
            node.parameters.push(ts.factory.createParameterDeclaration(
                undefined,undefined,undefined,ts.factory.createIdentifier('c')
            ))
            node.jsDoc = undefined
            break;
          }
        })

        console.log(printer.printFile(sourceFile));
      })

}
/**
 *
 * @param {ts.Node} node
 * @param {(node: ts.Node) => void} cbNode
 */
function travers(node,cbNode) {
  cbNode(node)
  ts.forEachChild(node,(_)=>travers(_,cbNode))
}
// Run the extract function with the script's arguments

extract()
```

### 🙁 Actual behavior
Console outputs

```ts
/**
 * 数字相加
 * @param  a 数字aa
 * @param  b 数字b
 * @returns  a+b的结果
 */
function add2(a: number, b: number, c): number {
    return a + b;
}
```

### 🙂 Expected behavior

```diff ts
- /**
-  * 数字相加
-  * @param  a 数字aa
-  * @param  b 数字b
-  * @returns  a+b的结果
-  */
function add2(a: number, b: number, c): number {
    return a + b;
}
```

**Comment by RyanCavanaugh** [comment]
These objects aren't intended to be mutated.

**Comment by Binote** [comment]
The AST should be able to describe all code in the file and also modify them.
Sorry, I can't agree with you
